### DISCUSSION TITLE
Deleting a Webspace

### DISCUSSION THREAD

**Discussion by brachycera** [sulu/sulu#6650] [Q&A] [answered]
I added two new Webspace XML files, then I set the Userrights for both new Webspaces.

Afterwards I deleted one of the Webspace XML files and re-run the following commands.

```bash
$ php bin/adminconsole cache:clear

$ php bin/adminconsole sulu:document:initialize

$ php bin/adminconsole sulu:content:validate:webspaces
```

Now there are still fragments of the deleted Webspace in the Database tables `se_user_settings` and `se_permissions`. Is that intended behavior or do I need to run an additional `bin/adminconsole` command to remove the orphaned Database entries?

**Comment by alexander-schranz** [comment, marked as answer]
There is currently no command which will cleanup your data from a removed or renamed webspace. You need to cleanup `se_user_settings` and `se_permissions` yourself. Mostly by creating a doctrine migration via the doctrine migration bundle. But you can also do it manually. Beside that two tables there is also a phpcr node which can be removed under `/cmf/<old_webspace_key>` you can remove it via a phpcr migration or via `bin/console doctrine:phpcr:node:remove` and `bin/websiteconsole doctrine:phpcr:node:remove` command.

As I never did something like that I can not be 100% sure if there is something else missing. But let us know. And a pull request to the docs maybe as part of https://docs.sulu.io/en/2.3/cookbook/create-new-webspace.html is welcome :)

**Comment by brachycera** [comment]
Thanks again @alexander-schranz for the lightning fast response!
I will try your suggestions, but I first need to make myself familiar with the [DoctrineMigrationsBundle](https://symfony.com/bundles/DoctrineMigrationsBundle/current/index.html). And how [PHPCR](https://phpcr.github.io/about/) works since I never worked with this **PHP Content Repository** before.
